tests/many_to_many/tests.py

```diff
@@ -554,3 +564,38 @@ def test_inherited_models_selects(self):
             ]
         )
         self.assertQuerysetEqual(b.publications.all(), ['<Publication: Science Weekly>'])
+
+    def test_count_query_optimization(self):
+        """
+        #29725 - Calling count() on a many to many relation
```

Comment:
```
"""
QuerySet.count() on a many-to-many relation doesn't include an
unnecessary JOIN.
"""
```
Ticket references should be reserved for obscure issues (not needed here, I think).